enhancements/api-review/gcp_user_defined_labels.md

```diff
@@ -0,0 +1,610 @@
+func() {
+	select {
+	case <-featureGateAccessor.InitialFeatureGatesObserved():
+		featureGates, err := featureGateAccessor.CurrentFeatureGates()
+	case <-time.After(1 * time.Minute):
+		log.Error(nil, "timed out waiting for FeatureGate detection")
+		return nil, fmt.Errorf("timed out waiting for FeatureGate detection")
+	}
+
+	gcpLabelsTagsEnabled := featureGates.Enabled(configv1.FeatureGateGCPLabelsTags)
+
+	if gcpLabelsTagsEnabled {
+		// add user defined labels and tags
+	}
+}
+```
+
+#### Caveats
+1. Updating or removing resource labels/tags added by OpenShift using an external interface,
+   may or may not be reconciled by the operator managing the resource.
+2. Updating labels/tags of individual resources is not supported and any label or tag present
+   in `.status.platformStatus.gcp.resourceLabels` or `.status.platformStatus.gcp.resourceTags`
+   respectively of `infrastructure.config.openshift.io/v1` resource will result in adding 
+   labels and tags to all OpenShift managed GCP resources.
+
+### Risks and Mitigations
+- Risks in using google-beta provider:
+
+  Except for the additional features/functionality available for preview in the beta version,
+  google-beta provider is not different from the GA version google provider. google-beta
+  provider binary is embedded into installer as part of the installer build process and there
+  are no additional requirements made on the end user to use the beta version.
+
+  Below are some of the risks in using the google-beta provider.
+  1. google-beta is a preview version and doesn't always carry the SLAs or technical
+  support obligations.
+  2. google-beta provider always uses the beta API regardless of whether the
+  request contains any beta features.
+  3. google_compute_address API provides just the `labels` functionality in the beta version,
+  which is in beta since the initial [commit](https://github.com/hashicorp/terraform-provider-google-beta/commit/0fdc262183ed82e78824e42d05042a7b0aba7c8b)
+  in 2018, though the usual beta phase for a feature is six months.
+
+### Drawbacks
+- User-defined labels cannot be updated on an GCP resource which is not managed by an
+  operator. In this proposal, the changes proposed and developed will be part of
+  openshift-* namespace. External operators are not in scope.
+  User-defined labels can be updated on the following GCP resources.
+  - Compute Engine Instance
+  - Compute Address
+  - Compute Image
+  - Compute Forwarding Rule
+  - Storage Bucket
+  - DNS Zone
+
+- GCP Labels can be used as queryable annotations for resources, but can't be used
+  to set conditions on policies.
+
+- User-defined tags cannot be updated on a GCP resource which is not managed by an
+  operator. In this proposal, the changes proposed and developed will be part of
+  openshift-* namespace. External operators are not in scope.
+  User-defined tags can be updated on the following GCP resources.
+  - Compute Engine Instances
+  - Cloud Storage Buckets
```

Comment:
If tags can only be created on two resources, this seems like a lot of effort to tag only these two. 

What is the motivation for adding tags in addition to labels? i.e. does enabling tags allow us to do something we can't do with just labels?

Reply:
If we need to tag buckets created for persistent volumes, which should get review from @jsafrane

Reply:
Tags provide additional functionalities of able to set IAM policy conditions, Organization policy conditions, cost analysis and integration with cloud billing, whereas labels can be used for organizing resources and integration with cloud billing.

> If we need to tag buckets created for persistent volumes, which should get review from @jsafrane

Thank you for the suggestion! I will reach out to @jsafrane for the review.